File: C++/349-intersection-of-two-arrays.cpp
#include<vector>
#include<unordered_set>

using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        std::unordered_set<int> results;
        std::unordered_set<int> nums(nums1.begin(), nums1.end());
        for (int num : nums2) {
            if (nums.find(num) != nums.end()) {
                results.insert(num);
            }
        }
        return std::vector<int> (results.begin(), results.end());
    }
};

File: JS/349-intersection-of-two-arrays.js
// set
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let cnt = new Set(nums1);
    let res = new Set();
    for (let num of nums2) {
        if (cnt.has(num)) res.add(num);
    }
    return Array.from(res);
};